### PR TITLE
LPS-169605 Do not update scheduler, delete and re-schedule instead

### PR DESCRIPTION
**Pull request by tinatian**

### CODE REVIEW

**Comment by tinatian** [comment]
ci:test:relevant

**Comment by tinatian** [comment]
ci:test:sf

**Comment by tinatian** [comment]
ci:test:relevant

**Comment by tinatian** [comment]
ci:test:sf

File: modules/apps/notification/notification-service/src/main/java/com/liferay/notification/internal/instance/lifecycle/NotificationQueuePortalInstanceLifecycleListener.java [hunk 1]
```diff
@@ -49,92 +50,133 @@
...
 	}
 
 	private static final Log _log = LogFactoryUtil.getLog(
 		NotificationQueuePortalInstanceLifecycleListener.class);
 
-	private BundleContext _bundleContext;
+	@Reference(
+		target = "(component.factory=com.liferay.notification.internal.messaging.CheckNotificationQueueEntryMessageListener)"
+	)
+	private ComponentFactory _componentFactory;
```
Comment (shuyangzhou) [inline modules/apps/notification/notification-service/src/main/java/com/liferay/notification/internal/instance/lifecycle/NotificationQueuePortalInstanceLifecycleListener.java]: ComponentFactory is a generic type.

File: modules/apps/notification/notification-service/src/main/java/com/liferay/notification/internal/instance/lifecycle/NotificationQueuePortalInstanceLifecycleListener.java [hunk 2]
```diff
@@ -49,92 +50,133 @@
...
+		}
+
+		private NotificationQueueConfigurationManagedServiceFactory() {
+			_defaultNotificationQueueConfiguration =
+				ConfigurableUtil.createConfigurable(
+					NotificationQueueConfiguration.class,
+					Collections.emptyMap());
+		}
+
+		private Map<String, Long> _companyIds = new ConcurrentHashMap<>();
```
Comment (shuyangzhou) [inline modules/apps/notification/notification-service/src/main/java/com/liferay/notification/internal/instance/lifecycle/NotificationQueuePortalInstanceLifecycleListener.java]: This should be final

File: modules/apps/notification/notification-service/src/main/java/com/liferay/notification/internal/instance/lifecycle/NotificationQueuePortalInstanceLifecycleListener.java [hunk 3]
```diff
@@ -49,92 +50,133 @@
...
+	)
+	private ComponentFactory _componentFactory;
+
+	private final Map<Long, ComponentInstance<?>> _componentInstances =
+		new ConcurrentHashMap<>();
 
 	@Reference
 	private ConfigurationProvider _configurationProvider;
 
 	@Reference(target = ModuleServiceLifecycle.PORTAL_INITIALIZED)
```
Comment (shuyangzhou) [inline modules/apps/notification/notification-service/src/main/java/com/liferay/notification/internal/instance/lifecycle/NotificationQueuePortalInstanceLifecycleListener.java]: This is not needed.

**Comment by tinatian** [comment]
ci:test:relevant

**Comment by tinatian** [comment]
ci:test:sf

**Pull request closed without merging**